src/libsyntax/parse/parser.rs

```diff
@@ -2243,9 +2243,9 @@ impl<'a> Parser<'a> {
                         attrs);
                 }
                 if self.is_catch_expr() {
+                    let lo = self.span;
```

Comment:
I changed from using `prev_span` to `span`, so we can move it before the eat(Do) expression